sys/sys/socket.h

```diff
@@ -427,17 +427,16 @@ struct sockproto {
  * Message header for recvmsg and sendmsg calls.
  * Used value-result for recvmsg, value only for sendmsg.
  */
-#ifndef _KERNEL
 struct msghdr {
-	void		*msg_name;		/* optional address */
+	void * __kerncap msg_name;		/* optional address */
 	socklen_t	 msg_namelen;		/* size of address */
-	struct iovec	*msg_iov;		/* scatter/gather array */
+	struct iovec * __kerncap msg_iov;	/* scatter/gather array */
 	int		 msg_iovlen;		/* # elements in msg_iov */
-	void		*msg_control;		/* ancillary data, see below */
+	void * __kerncap msg_control;		/* ancillary data, see below */
 	socklen_t	 msg_controllen;	/* ancillary data buffer len */
 	int		 msg_flags;		/* flags on received message */
 };
-#else /* _KERNEL */
+#ifdef _KERNEL
 #if __has_feature(capabilities)
 struct msghdr_c {
```

Comment:
As @bsdjhb noted for iovec, we probably want to move msghdr in the respective compat headers. This would be consistent with what is being done for freebsd32.